src/validator.js

```diff
@@ -75,6 +88,44 @@ export default class Validator {
     return output;
   }
 
+  scan(_Xpi=Xpi) {
+    return new Promise((resolve, reject) => {
+      this.checkFileExists(this.packagePath)
+        .then(() => {
+          this.xpi = new _Xpi(this.packagePath);
+          return this.xpi.getMetaData();
+        }).then(() => {
+          return this.xpi.getJSFiles();
+        })
+        .then((jsFiles) => {
+          jsFiles.forEach((filename) => {
+            this.xpi.getFileAsString(filename)
+              .then((code) => {
+                let jsScanner = new JavaScriptScanner(code, filename);
+                return jsScanner.scan();
+              })
+              .then((validatorMessages) => {
+                validatorMessages.forEach((message) => {
+                  this.addToCollector(message);
+                });
+
+                resolve(this.collector);
```

Comment:
Doesn't this mean you'll resolve on the first file - what about the following iterations.